src/response.rs

```diff
@@ -134,6 +151,48 @@ impl<'de> Deserialize<'de> for PeriodTime {
     }
 }
 
+struct MetricsHierarchyVisitor;
+
+impl<'de> Visitor<'de> for MetricsHierarchyVisitor {
+    type Value = MetricsHierarchy;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter.write_str("a structure that represents a hierarchy of metrics")
+    }
+
+    fn visit_map<V>(self, mut map: V) -> Result<Self::Value, V::Error>
+    where
+        V: MapAccess<'de>,
+    {
+        let mut hierarchy = MetricsHierarchy::new();
+
+        // The key in the hierarchy structure is always "metric". It is not
+        // used, but we need to read it to get the value.
+        let mut key: Option<String> = map.next_key()?;
+        while key != None {
```

Comment:
maybe `while map.next_key()?.is_some() {` or `while let Some(_) = map.next_key()? {` and you can remove the other two lines referencing `key`?

Reply:
Those do not work as they are. There are some missing type annotations.

Reply:
Fixed :+1: